widget/radio.go

```diff
@@ -276,8 +279,19 @@ func (r *Radio) CreateRenderer() fyne.WidgetRenderer {
 	return &radioRenderer{items, objects, r}
 }
 
+// SetMandatory sets if the radio must have a selection.
+func (r *Radio) SetMandatory(m bool) {
+	r.Mandatory = m
+	if r.Selected == "" && len(r.Options) > 0 {
```

Comment:
Would it not be better to return from this and do nothing rather than resetting it to a different value?

Reply:
Then you would not need to have `SetMandatory` at all :).
I wanted the `Radio` to ensure a valid option is selected even if I don't pre-select one.
I use it for dark/light theme selection on my settings page and really don't care which one is pre-selected.

A mandatory `Radio` does not allow to deselect an option. I think it is a pragmatic approach to simply chose one if the `Radio` is made mandatory and no option has been selected. I don't wanted to add form input validation (because I don't need it ;)).

In the light of future input validation it might be better to not pre-select automatically.
Therefore I would remove `SetMandatory` again, okay?